[tensorexpr] Simplify x/100 -> 0 if x is a non-negative integer less than 100.

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64763

Simplification pattern:
  x/N -> 0; N is a constant positive integer and x is a for-loop index whose range is a subset of [0, N).

Differential Revision: [D30845854](https://our.internmc.facebook.com/intern/diff/D30845854)

huiguoo committed:
[tensorexpr] Simplify x/100 -> 0 if x is a non-negative integer less than 100.

  Simplification pattern:
  x/N -> 0; N is a constant positive integer and x is a for-loop index whose range is a subset of [0, N).

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30845854).

bertmaher commented on torch/csrc/jit/tensorexpr/ir_simplifier.cpp:
```diff
@@ -2881,6 +2881,29 @@ ExprPtr SimplifierUnderContext::mutate(DivPtr v) {
+          IRSimplifier::simplify(alloc<CompareSelect>(end, rhsScalar, kLE));
+      if (check_start->isConstant() && check_end->isConstant() &&
+          immediateEquals(check_start, 1) && immediateEquals(check_end, 1)) {
+        oss << "SimplifierUnderContext: " << *v << " => " << immLike(lhsVar, 0)
+            << "\n";
+        GRAPH_DEBUG(oss.str());
```
Rewrite the debug logging as
```suggestion
        GRAPH_DEBUG("SimplifierUnderContext: ", *v, " => ", immLike(lhsVar, 0));
```
so that we don't construct a stringstream unless debug logging is requested.

huiguoo committed:
Update on "[tensorexpr] Simplify x/100 -> 0 if x is a non-negative integer less than 100."


Simplification pattern:
  x/N -> 0; N is a constant positive integer and x is a for-loop index whose range is a subset of [0, N).

Differential Revision: [D30845854](https://our.internmc.facebook.com/intern/diff/D30845854)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30845854).

huiguoo committed:
Update on "[tensorexpr] Simplify x/100 -> 0 if x is a non-negative integer less than 100."


Simplification pattern:
  x/N -> 0; N is a constant positive integer and x is a for-loop index whose range is a subset of [0, N).

Differential Revision: [D30845854](https://our.internmc.facebook.com/intern/diff/D30845854)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30845854).
